infra/network/src/main/java/com/simprints/infra/network/url/BaseUrlProviderImpl.kt

```diff
@@ -26,13 +28,28 @@ internal class BaseUrlProviderImpl @Inject constructor(
             "https://${BuildConfig.BASE_URL_PREFIX}.simprints-apis.com$BASE_URL_SUFFIX"
     }
 
-    val prefs: SharedPreferences = context.getSharedPreferences(PREF_FILE_NAME, PREF_MODE)
+    @Deprecated("Data has been migrated to secure prefs. Delete after there are no users below 2025.3.0")
+    private val prefs: SharedPreferences = context.getSharedPreferences(LEGACY_PREF_FILE_NAME, Context.MODE_PRIVATE)
+    private val securePrefs = securityManager.buildEncryptedSharedPreferences(SECURE_PREF_FILE_NAME)
 
-    override fun getApiBaseUrl(): String = prefs
+    /**
+     * Ensures that data has been migrated to secure prefs before accessing it.
+     */
+    private fun getSecurePrefs(): SharedPreferences {
+        if (prefs.contains(API_BASE_URL_KEY)) {
```

Comment:
Instead of checking the old prefs (and always reading 2 files as a consequence) would it make sense to first check the securePrefs for the flag and read old prefs only if it's missing?